lib/MusicBrainz/Server/Controller/WS/2/Genre.pm

```diff
@@ -0,0 +1,70 @@
+package MusicBrainz::Server::Controller::WS::2::Genre;
+use Moose;
+use MusicBrainz::Server::Validation qw( is_guid );
+
+BEGIN { extends 'MusicBrainz::Server::ControllerBase::WS::2' }
+
+use aliased 'MusicBrainz::Server::WebService::WebServiceStash';
+
+my $ws_defs = Data::OptList::mkopt([
+    genre => {
+        method   => 'GET',
+        required => [ qw(query) ],
+        optional => [ qw(fmt limit offset) ],
+    },
+    genre => {
+        method   => 'GET',
+        optional => [ qw(fmt limit offset) ],
+    },
+    genre => {
+        method   => 'GET',
+        optional => [ qw(fmt) ],
+    }
+]);
+
+with 'MusicBrainz::Server::WebService::Validator' => {
+    defs => $ws_defs,
+};
+
+with 'MusicBrainz::Server::Controller::WS::2::Role::Lookup' => {
+    model => 'Genre',
+};
+
+sub base : Chained('root') PathPart('genre') CaptureArgs(0) { }
+
+sub genre_toplevel {
+    my ($self, $c, $stash, $genres) = @_;
+
+}
+
+sub genre_browse : Private {
+    my ($self, $c) = @_;
+
+    my ($resource, $id) = @{ $c->stash->{linked} };
+    my ($limit, $offset) = $self->_limit_and_offset($c);
+
+    if (!is_guid($id)) {
+        $c->stash->{error} = "Invalid mbid.";
+        $c->detach('bad_req');
+    }
+
+    my $genres;
```

Comment:
This doesn't appear to be initialized anywhere, but it's being accessed/passed around twice below (even `$genres->{items}`, which Perl allows, but).